Modules/FindCUDA.cmake

```diff
@@ -683,6 +684,24 @@ find_path(CUDA_TOOLKIT_INCLUDE
 find_path(CUDA_TOOLKIT_INCLUDE device_functions.h)
 mark_as_advanced(CUDA_TOOLKIT_INCLUDE)
 
+if (CUDA_VERSION VERSION_GREATER "7.0")
+  set(cuda_fp16_found ${CUDA_TOOLKIT_INCLUDE})
+else()
+  find_path( cuda_fp16_found
+    cuda_fp16.h # Header included in toolkit
+    PATHS ${CUDA_TOOLKIT_INCLUDE}
+    ENV CUDA_PATH
+    ENV CUDA_INC_PATH
+    NO_DEFAULT_PATH
+    )
+endif()
+
+if (cuda_fp16_found)
+  set(CUDA_HAS_FP16 true CACHE BOOL "CUDA has half-float support")
+else()
+  set(CUDA_HAS_FP16 false CACHE BOOL "CUDA has half-float support")
+endif()
```

Comment:
I don't think the `CUDA_HAS_FP16` result variable needs to be cached because it can be re-computed from the other results on every run.  It is not something we expect the user to set, is it?